src/sparsezoo/v2/objects/model_directory.py

```diff
@@ -377,32 +395,56 @@ def _file_from_files(
 
         if not files_found:
             return None
-        if len(files_found) == 1:
+        elif len(files_found) == 1:
+            return files_found[0]
+
+        elif display_name == "model.onnx" and len(files_found) == 2:
+            # `model.onnx` file may be found twice:
+            #   - directly in the root directory
+            #   - inside `deployment` directory
+            # if this is the case, return the first file
+            # (the other one is a duplicate)
             return files_found[0]
         else:
             return files_found
 
     def _directory_from_files(
         self,
         files: List[Dict[str, Any]],
-        directory_class: Union[Directory, NumpyDirectory, SampleOriginals] = Directory,
+        directory_class: Union[Directory, NumpyDirectory] = Directory,
         display_name: Optional[str] = None,
-        regex: Optional[bool] = True,
+        regex: Optional[bool] = False,
         allow_multiple_outputs: Optional[bool] = False,
     ) -> Union[Directory, None]:
+
         # Takes a list of file dictionaries and returns
         # a Directory() object, if successful,
         # otherwise None.
-        directories_found = []
-        for file in files:
-            directory = self._get_directory(
-                file=file,
-                directory_class=directory_class,
-                display_name=display_name,
-                regex=regex,
+        if all([file_dict.get("file_type") for file_dict in files]):
```

Comment:
Same old favorite nit to take advantage of `all`

```suggestion
        if all(file_dict.get("file_type") for file_dict in files):
```